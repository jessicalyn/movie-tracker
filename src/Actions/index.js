export const addMovies = (movies) => ({
  type: "ADD_MOVIES",
  movies
})

export const addUser = (user) => ({
  type: "ADD_USER",
  user
  
})

export const loginUser = (email) => ({
  type: "LOGIN_USER",
  email
})

