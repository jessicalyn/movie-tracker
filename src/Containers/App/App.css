.App {
  text-align: center;
}

h1 {
  font-family: 'Limelight', cursive;
  font-size: 3rem
}

.App-header{
  display: flex;
}
