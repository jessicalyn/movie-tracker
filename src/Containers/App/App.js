import React, { Component } from 'react';
import './App.css';
import { fetchData } from '../../Utils/fetchData'
import Movies from '../Movies/Movies'
import { APIkey } from '../../Utils/APIkey';
import { addMovies } from '../../Actions/index'
import { connect } from 'react-redux'
import { NavLink, Route } from 'react-router-dom'
import { Login } from '../../Components/Login'

export class App extends Component {

  componentDidMount = async () => {
    this.fetchMovies()
  }

  fetchMovies = async () => {
    const movies = await fetchData(APIkey)
    this.props.addMovies(movies.results)
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1>Movie Tracker</h1>
          <NavLink to="/login" className="nav">Login</NavLink>
        </header>
        <Route exact path='/' component={Movies} />
        <Route exact path='/login' component={Login} />
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies) => dispatch(addMovies(movies))
})

export default connect(null, mapDispatchToProps)(App);


