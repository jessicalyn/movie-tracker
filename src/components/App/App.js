import React, { Component } from 'react';
import './App.css';
import { fetchData } from '../../Utils/fetchData'
import Movies from '../Movies/Movies'
import { APIkey } from '../../Utils/APIkey'

class App extends Component {
  constructor(){
    super()
    this.state = {
      movies: []
    }
  }

  componentDidMount = async () => {
    const fetchMovies = await fetchData(APIkey)
    this.setState({ movies: fetchMovies.results})
  }

  render() {
    const { movies } = this.state
    return (
      <div className="App">
        <header className="App-header">
          <h1>Movie Tracker</h1>
        </header>
        <Movies movies={movies} />
      </div>
    );
  }
}

export default App;
