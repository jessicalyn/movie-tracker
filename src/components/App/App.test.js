import React from 'react';
import ReactDOM from 'react-dom';
import { App, mapDispatchToProps } from './App'
import { addMovies } from '../../Actions'

describe('App', () => {

  describe('mapDispatchToProps', () => {
    it('should call dispatch when using a function from mapDispatchToProps', () => {
      const mockDispatch = jest.fn()
      const mockMovies = [
        {title: "Cars", poster_path: "car.jpg"},
        {title: "How To Train Your Dragon", poster_path: "dragon.jpg"}
      ]
      const actionToDispatch = addMovies(mockMovies)
      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.addMovies(mockMovies)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})