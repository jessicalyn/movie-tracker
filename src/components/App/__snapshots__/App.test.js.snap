// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App it should match the snapshot with all data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
        >
          <h1>
            Movie Tracker
          </h1>
        </header>,
        <Connect(Movies) />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Movie Tracker
          </h1>,
          "className": "App-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Movie Tracker",
          },
          "ref": null,
          "rendered": "Movie Tracker",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
          >
            <h1>
              Movie Tracker
            </h1>
          </header>,
          <Connect(Movies) />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Movie Tracker
            </h1>,
            "className": "App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Movie Tracker",
            },
            "ref": null,
            "rendered": "Movie Tracker",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
